### PR TITLE
fill in ECS-compat top-level fields from context

### PR DESCRIPTION
**Pull request by graphaelli**
* agent <- context.service.agent
* host <- context.system
* process <- context.process
  * process.args <- context.process.argv, now indexed
* service <- context.service

Also indexes observer.listening, missed previously.

### CODE REVIEW

File: _meta/fields.common.yml [hunk 1]
```diff
@@ -324,3 +262,89 @@
...
+          description: >
+            IP of the host that records the event.
+          overwrite: true
+
+    - name: process
+      type: group
+      description: >
+        Information pertaining to the running process where the data was collected
+      fields:
+        - name: args
```
[the remark this answers is not in the document]
Comment (jalvz) [reply _meta/fields.common.yml]: oh, so this is going to be indexed then?

**Comment by jalvz** [comment]
Hmm, i thought it was because of the onboarding document ordering

**Comment by simitt** [comment]
It is because the metadata are moved out of `context`, so the test needs to be adapted. I am on it.

**Comment by simitt** [comment]
@graphaelli I did a `force-push` because I rebased to master.

File: _meta/fields.common.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (graphaelli) [reply _meta/fields.common.yml]: Some agents omit this field by default so I don't think we should be too concerned with the extra space used.

File: _meta/fields.common.yml [hunk 2]
```diff
@@ -324,3 +262,89 @@
...
+        - name: version
+          type: keyword
+          description: >
+            Version of the agent used.
+          overwrite: true
+
+    - name: host
+      type: group
+      description: >
+        Optional system fields.
```
[the remark this answers is not in the document]
Comment (graphaelli) [reply _meta/fields.common.yml]: done

File: model/metricset/event.go [hunk 3]
```diff
@@ -144,20 +144,21 @@ func (me *Metricset) Transform(tctx *transform.Context) []beat.Event {
...
-		context["tags"] = me.Tags
+		tags := common.MapStr{}
+		// normalize map entries by calling utility.Add
+		utility.Add(tags, "tags", me.Tags)
+		fields.Put("context", tags)
 	}
-
-	fields["context"] = tctx.Metadata.Merge(context)
-	fields["processor"] = processorEntry
+	tctx.Metadata.MergeMinimal(fields)
```
[the remark this answers is not in the document]
Comment (graphaelli) [reply model/metricset/event.go]: Great catch, this was not intentional.  Will add a test to catch this regression and fix it.

[the remark this answers is not in the document]
Comment (graphaelli) [reply model/metricset/event.go]: This should have been caught while I updated the approval docs.

**Comment by graphaelli** [comment]
Thanks for the fixups.  All review feedback has been addressed and one test has been skipped, @simitt is already working on that one, thank you.

File: processor/stream/package_tests/metadata_attrs_test.go [hunk 4]
```diff
@@ -95,11 +97,15 @@ func getMetadataEventAttrs(t *testing.T, prefix string) *tests.Set {
 	return eventFields
 }
 
+func metadataFieldsNotInPayloadAttrs() *tests.Set {
+	return tests.NewSet("context.process.args")
+}
+
 func TestMetadataPayloadAttrsMatchFields(t *testing.T) {
+	t.Skip("temporarily disabled")
```
[the remark this answers is not in the document]
Comment (graphaelli) [reply processor/stream/package_tests/metadata_attrs_test.go]: thanks!

**Comment by graphaelli** [comment]
only the sync job failed, as expected.

**Pull request merged**